docs/standard/io/file-path-formats.md

```diff
@@ -0,0 +1,170 @@
+- An optional filename. The [directory separator character](<xref:System.IO.Path.DirectorySeparatorChar>) separates the file path and the filename.
+
+The following are some examples of UNC paths:
+
+|Path  |Description  |
+| -- | -- |
+| \\\\system07\\C$\\ | The root directory of the C: drive on `system07`. |
+| \\Server2\\Share\\Test\\Foo.txt | The Foo.txt file in the Test directory of the \\\\Server2\\Share volume.|
+
+UNC paths are always absolute. You can use relative paths only by mapping a UNC path to a drive letter.
+
+## DOS device paths
+
+The Windows operating system has a unified object model that points to all resources, including files. These object paths are not directly accessible from the Windows APIs (such as the Console windows and Windows Explorer). They are, however, exposed to the Win32 layer through a special folder of symbolic links that legacy DOS and UNC paths are mapped to. This special folder is accessed via the DOS device path syntax, which is one of:
+
+\\.\C:\Test\Foo.txt   
+\\?\C:\Test\Foo.txt
+
+> [!NOTE]
+> DOS device path syntax is supported on .NET implementations running on Windows starting with .NET Core 1.1 and .NET Framework 4.6.2.
+
+The DOS device path consists of the following components:
+
+- `\\.\` or `\\?\`, which identifies the path as a DOS device path.
+- A symbolic link to the "real" device object (C: in this case). 
+
+   There is a specific link for UNCs that is called, not surprisingly, `UNC`. For example:
+
+      \\.\UNC\Server\Share\Test\Foo.txt   
+      \\?\UNC\Server\Share\Test\Foo.txt
+
+Like UNCs, DOS device paths are fully qualified by definition. Current directories never enter into their usage.
+
+## Example: Ways to refer to the same file
+
+The following example illustrates some of the ways in which you can refer to a file when using the APIs in the <xref:System.IO> namespace. The example instantiates a <xref:System.IO.FileInfo> object and uses its <xref:System.IO.FileInfo.Name> and <xref:System.IO.FileInfo.Length> properties to display the filename and the length of the file.
+
+[!code-csharp[referring-to-the-same-file](~/samples/snippets/standard/io/file-names/cs/file-refs.cs)]
+[!code-vb[referring-to-the-same-file](~/samples/snippets/standard/io/file-names/vb/file-refs.vb)]
+
+## Path normalization
+
+Almost all paths passed to Windows APIs are normalized. During normalization, Windows performs the following steps:
+
+- Identifies the path.
+- Applies the current directory to partially qualified (relative) paths.
+- Canonicalizes component and directory separators.
+- Evaluates relative directory components (`.` for the current directory and `..` for the parent directory).
+- Trims certain characters.
+
+This normalization happens implicitly, but you can do it explicitly by calling the Windows [GetFullPathName() function](https://msdn.microsoft.com/library/windows/desktop/aa364963(v=vs.85).aspx).
```

Comment:
Are you suggesting to use PInvoke here? If you are, maybe make that more explicit? For example:

> This normalization happens implicitly, but you can do it explicitly by calling
the Windows [GetFullPathName() function](https://msdn.microsoft.com/library/windows/desktop/aa364963(v=vs.85).aspx) using P/Invoke.

Reply:
You can also do this by calling `Path.GetFullPath()`.  `Path.GetFullPath()` is the .NET wrapper for `GetFullPathName()`.